File: projekty/PO_Projekt2/ProgOb_Projekt2/src/main/java/pl/Dybowski/Adrian/Projekt2/WirtualnySwiat/Rosliny/BarszczSosnowskiego.java
package pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Rosliny;

import java.util.HashSet;

import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Organizm;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.RodzajeKomunikatow;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Stale;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Swiat;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Wspolrzedne2D;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Zwierzeta.Zwierze;

public class BarszczSosnowskiego extends Roslina {

	public BarszczSosnowskiego(Swiat swiat, Wspolrzedne2D polozenie) {
		super(swiat, polozenie, Stale.sila.get(BarszczSosnowskiego.class), Stale.inicjatywa.get(BarszczSosnowskiego.class),
				Stale.szansaRozprzestrzeniania.get(BarszczSosnowskiego.class), Stale.probyRozprzestrzeniania.get(BarszczSosnowskiego.class));
	}
	
	@Override
	public void akcja() {
		if (!zyje) return;
		
		HashSet<Wspolrzedne2D> cele = swiat.sasiedniePola(polozenie);
		
		for (Wspolrzedne2D cel : cele) {
			if (!swiat.poleZajete(cel)) continue;
			
			if (swiat.organizmNaPolu(cel) instanceof Zwierze) {
				swiat.dodajKomunikat(this, swiat.organizmNaPolu(cel), RodzajeKomunikatow.zabija);
				swiat.organizmNaPolu(cel).zakonczZycie();
			}
		}
		
		super.akcja();
	}
	
	@Override
	public void kolizja(Organizm ofiara, boolean atak) {
		if (atak) return;
		
		if (ofiara.isZyje()) ofiara.zakonczZycie();
		
		if (zyje) zakonczZycie();
	}

	@Override
	public boolean tegoSamegoTypu(Organizm inny) {
		return inny instanceof BarszczSosnowskiego;
	}

	@Override
	public Organizm stworzDziecko(Wspolrzedne2D cel) {
		return new BarszczSosnowskiego(swiat, cel);
	}

}

File: PO_Projekt3/barszczsosnowskiego.py
from roslina import Roslina
from zwierze import Zwierze

class BarszczSosnowskiego(Roslina):
    def __init__(self, swiat, polozenie):
        super().__init__(swiat, polozenie, 10, 0, 0.0, 1)

    def akcja(self):
        if not self._zyje:
            return
        
        cele = self._swiat.sasiedniePola(self._polozenie)

        for cel in cele:
            from cyberowca import CyberOwca
            if self._swiat.poleZajete(cel) and isinstance(self._swiat.organizmNaPolu(cel), Zwierze) and not isinstance(self._swiat.organizmNaPolu(cel), CyberOwca):
                self._swiat.dodajKomunikat(str(self._polozenie) + ": " + self.nazwaKlasy() + " zabija " + self._swiat.organizmNaPolu(cel).nazwaKlasy())
                self._swiat.organizmNaPolu(cel).zakonczZycie()

        super().akcja()

    def kolizja(self, ofiara, atak):
        if atak:
            return

        from cyberowca import CyberOwca
        if ofiara._zyje and not isinstance(ofiara, CyberOwca):
            ofiara.zakonczZycie()

        if self._zyje:
            self.zakonczZycie()

    def tegoSamegoTypu(self, inny):
        return isinstance(inny, BarszczSosnowskiego)

    def stworzDziecko(self, cel):
        return BarszczSosnowskiego(self._swiat, cel)

    def rysowanie(self):
        return (0, 77, 0)
